### PR TITLE
Transformer kernel/fix layer norm

### PR DESCRIPTION
**Pull request by RezaYazdaniAminabadi**
This PR fixes https://github.com/microsoft/DeepSpeed/issues/1563

### CODE REVIEW

File: deepspeed/inference/engine.py [hunk 1]
```diff
@@ -227,6 +227,8 @@ def forward(self, *inputs, **kwargs):
             *inputs: Variable length input list
             **kwargs: variable length keyword arguments
         """
+        import pdb
```
Comment (tjruwase) [inline deepspeed/inference/engine.py:230]: Remove debugger code?

File: deepspeed/ops/transformer/inference/transformer_inference.py [hunk 2]
```diff
@@ -146,6 +146,31 @@ def _transpose_for_context(x):
                                       (hidden_size_per_partition,)
             return x.view(*new_x_layer_shape)
 
+        def backup_attention(mixed_query, key_layer, value_layer, input_mask):
```
Comment (tjruwase) [inline deepspeed/ops/transformer/inference/transformer_inference.py:149]: Is this part of the fix?

File: deepspeed/inference/engine.py (hunk 1 above)
Comment (RezaYazdaniAminabadi) [reply deepspeed/inference/engine.py:230]: thanks, I will remove it

File: deepspeed/ops/transformer/inference/transformer_inference.py (hunk 2 above)
Comment (RezaYazdaniAminabadi) [reply deepspeed/ops/transformer/inference/transformer_inference.py:149]: no it is not, not sure how it got there! I will remove it

File: csrc/transformer/normalize_kernels.cu [hunk 3]
```diff
@@ -1069,8 +1069,8 @@ __global__ void LayerNormBackward2(const float* out_grad,
 
     int row = blockIdx.x;
     int id = threadIdx.x;
-    int wid = id / WARP_SIZE;
-    int warp_num = (THREADS < row_stride ? THREADS : row_stride) / WARP_SIZE;
+    int wid = id >> 5;
```
Comment (tjruwase) [inline csrc/transformer/normalize_kernels.cu:1072]: Please replace the `5` with a descriptively named variable. Thanks.

**Review by tjruwase — APPROVED** [review]

File: csrc/transformer/normalize_kernels.cu (hunk 3 above)
Comment (RezaYazdaniAminabadi) [reply csrc/transformer/normalize_kernels.cu:1072]: got it. thanks

**Review by tjruwase — APPROVED** [review]

**Pull request merged**